src/rust/ide/src/model/execution_context/synchronized.rs

```diff
@@ -89,14 +112,27 @@ impl ExecutionContext {
     }
 
     /// Handles the update about expressions being computed.
-    pub fn handle_expression_updates
-    (&self, notification:ExpressionUpdates) -> FallibleResult {
-        self.model.computed_value_info_registry.apply_updates(notification.updates);
+    pub fn handle_notification
+    (&self, notification: Notification) -> FallibleResult {
+        match notification {
+            Notification::Completed => {
+                if !self.model.is_ready.replace(true) {
+                    WARNING!("Context {self.id} Became ready");
+                }
+            }
+            Notification::ExpressionUpdates(updates) => {
+                self.model.computed_value_info_registry.apply_updates(updates);
+            }
+        }
```

Comment:
If it is the ExecutionContext which both defines and handles the notification, why we cannot just create functions "handle_expression_updates" and "mark_as_ready"?